dall-e-shell.el

```diff
@@ -419,6 +421,34 @@ ERROR-CALLBACK otherwise."
                                        "KEY-NOT-FOUND")))))
                 "-d" (shell-maker--json-encode request-data))))
 
+(defun dall-e-shell-generate-image (prompt text)
+  "Generate image from text using DALL-E given prompt."
+  (let* ((question (concat prompt text))
+         (png-output (dall-e-shell-post-prompt question)))
+    png-output))
+
+(defun dall-e-shell-generate-image-from-region-0 (prompt)
+  "Generate image from region using DALL-E given prompt."
+  (let* ((text (buffer-substring-no-properties (region-beginning) (region-end)))
+         (png-output (dall-e-shell-generate-image prompt text)))
+    ;; if the current buffer is org-mode, insert a org-mode image link; if it is markdown mode, insert a markdown image link; otherwise, just insert the image
```

Comment:
Nice comment, though the code is already fairly simple and descriptive. We can do without it.